dev/devicelab/lib/framework/running_processes.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+@immutable
+class RunningProcessInfo {
+  const RunningProcessInfo(this.pid, this.creationDate, this.commandLine)
+      : assert(pid != null),
+        assert(commandLine != null);
+
+  final String commandLine;
+  final int pid;
+  final DateTime creationDate;
+
+  @override
+  bool operator ==(Object other) {
+    return other is RunningProcessInfo && other.pid == pid;
```

Comment:
technically should include commandLine and creationDate as well

Reply:
My thought is that we know the pid will always be the real key for this, and I don't have access to `hashValues` from dart:ui.  I'd rather not implement a proper hash for this and I know this way will work for the use case.

How about I just document it?

Reply:
Down the line you run the risk of someone else adding to this code and getting some _really_ weird error cases. Otherwise just use `foo.pid == bar.pid`

Reply:
Fair enough - updated.